### PR TITLE
language-server test and lint configuration

### PR DESCRIPTION
**Pull request by jackson-dean**

### CODE REVIEW

**Comment by jackson-dean** [comment]
there error is due to missing a test file. i'll update this with a simple test.

**Comment by jackson-dean** [comment]
I am still working on tests, but I think I have things in a more readable/manageable state. up to you, but my personal feeling is this is  fine to merge if you want to play around with it.

**Comment by jackson-dean** [comment]
Ok, I dug around through the vscode language serve code to see how they are testing it. It is actually a pretty nice way and was able to get some high level acceptance style tests. I think it should be good as a sanity check. But definitely will need to follow up with more unit tests and maybe a couple more sanity check acceptance tests. But i think what is here gives some confidence that things more or less work together properly.

**Comment by jackson-dean** [comment]
@chriseppstein let me know what you think when you get a chance to take a look.

File: packages/@css-blocks/vscode/src/extension.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ let client: LanguageClient;
 
 export function activate(_context: ExtensionContext) {
 	// The server is implemented in node
-	let serverModule = require.resolve('@css-blocks/language-server/out/server.js');
+	let serverModule = require.resolve('@css-blocks/language-server/dist/run.js');
 	let exists = fs.existsSync(serverModule);
 	console.log({exists});
```
Comment (chriseppstein) [inline packages/@css-blocks/vscode/src/extension.ts]: we can remove this logging that I accidently left in.

File: packages/@css-blocks/language-server/src/GlimmerAstPath.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  get node(): AST.Node {
+    return this.path[this.index];
+  }
+
+  get parent(): AST.Node | undefined {
+    return this.path[this.index - 1];
+  }
+
+  get parentPath(): GlimmerAstPath | undefined {
+    return new GlimmerAstPath(this.path, this.index - 1);
```
[the remark this answers is not in the document]
Comment (jackson-dean) [reply packages/@css-blocks/language-server/src/GlimmerAstPath.ts]: latest refactor there is no longer a class and no more native getters. it is just a plain function that returns a linked list of the "focus path" nodes.

File: packages/@css-blocks/language-server/src/test/util/createTextDocumentMock.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+interface LanguageIds {
+  [key: string]: string;
+}
+
+const extensionToLanguageMap: LanguageIds = {
+  js: "javascript",
+  ts: "typescript",
+  hb: "handlebars",
```
[the remark this answers is not in the document]
Comment (jackson-dean) [reply packages/@css-blocks/language-server/src/test/util/createTextDocumentMock.ts]: ahh, yes good catch. it is not currently used by anything, but it should be correct.

File: packages/@css-blocks/language-server/src/util/createFocusPath.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    if (containsPosition(range, position)) {
+      astPathNode.parent = {
+        data: astPathNode.data,
+        parent: astPathNode.parent,
+      };
+      astPathNode.data = node;
+    } else {
+      return null;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (jackson-dean) [reply packages/@css-blocks/language-server/src/util/createFocusPath.ts]: it is the case that some keys that are iterated do not represent nodes proper, but are arrays of nodes or keys that themselves contain arrays of nodes that need to be iterated through. I added a comment just after this to describe the situation.

**Pull request merged**